service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ApproveDraftOrdersService.java

```diff
@@ -0,0 +1,378 @@
+            counter++;
+        }
+
+        if (ordersToBeSent.isEmpty()) {
+            data.remove(ORDERS_TO_BE_SENT);
+        } else {
+            data.put(ORDERS_TO_BE_SENT, ordersToBeSent);
+        }
+
+        updateHearingDraftOrdersBundle(caseData, selectedOrdersBundle);
+        data.put("orderCollection", reviewedOrders);
+        data.put("hearingOrdersBundlesDrafts", caseData.getHearingOrdersBundlesDrafts());
+    }
+
+    private void updateHearingDraftOrdersBundle(CaseData caseData, Element<HearingOrdersBundle> selectedOrdersBundle) {
+        if (selectedOrdersBundle.getValue().getOrders().isEmpty()) {
+            caseData.getHearingOrdersBundlesDrafts()
+                .removeIf(bundle -> bundle.getId().equals(selectedOrdersBundle.getId()));
+        } else {
+            caseData.getHearingOrdersBundlesDrafts().stream()
+                .filter(bundle -> bundle.getId().equals(selectedOrdersBundle.getId()))
+                .forEach(bundle -> bundle.getValue().setOrders(selectedOrdersBundle.getValue().getOrders()));
+        }
+    }
+
+    private Element<HearingOrder> buildSealedHearingOrder(
+        ReviewDecision reviewDecision, Element<HearingOrder> hearingOrderElement) {
+        DocumentReference order;
+
+        if (JUDGE_AMENDS_DRAFT.equals(reviewDecision.getDecision())) {
+            order = reviewDecision.getJudgeAmendedDocument();
+        } else {
+            order = hearingOrderElement.getValue().getOrder();
+        }
+
+        return element(hearingOrderElement.getId(), hearingOrderElement.getValue().toBuilder()
+            .dateIssued(time.now().toLocalDate())
+            .status(CMOStatus.APPROVED)
+            .order(documentSealingService.sealDocument(order))
+            .lastUploadedOrder(order)
+            .build());
+    }
+
+    private Element<HearingOrder> buildRejectedHearingOrder(Element<HearingOrder> cmo, String changesRequested) {
+        return element(cmo.getId(), cmo.getValue().toBuilder()
+            .status(CMOStatus.RETURNED)
+            .requestedChanges(changesRequested)
+            .build());
+    }
+
+    private void updateHearingCMO(CaseData caseData, UUID cmoId) {
+        defaultIfNull(caseData.getHearingDetails(), new ArrayList<Element<HearingBooking>>()).stream()
+            .filter(hearing -> cmoId.equals(hearing.getValue().getCaseManagementOrderId()))
+            .findFirst()
+            .ifPresent(h -> h.getValue().setCaseManagementOrderId(null));
+    }
+
+    private Element<GeneratedOrder> buildBlankOrder(CaseData caseData, Element<HearingOrder> sealedOrder) {
+        HearingOrder order = sealedOrder.getValue();
+
+        return element(sealedOrder.getId(), GeneratedOrder.builder()
+            .type(BLANK_ORDER.getLabel())
+            .title(order.getTitle())
+            .document(order.getOrder())
+            .dateOfIssue(order.getDateIssued() != null ? formatLocalDateToString(order.getDateIssued(), DATE) : null)
+            .judgeAndLegalAdvisor(
```

Comment:
Maybe this should be the judge from the hearing for that order (if there is a different judge for that hearing) - might be worth checking with business. It is not always allocated judge

Reply:
updated to use the allocated judge from the hearing